msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/MultipleAccountPublicClientTests.java

```diff
@@ -0,0 +1,539 @@
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+                future.setResult(new AsyncResult<List<IAccount>>(null, exception));
+            }
+        });
+
+        RoboTestUtils.flushScheduler();
+
+        // assert
+        int expectedTenantProfilesCount = homeAccountSignedIn ?
+                testCaseData.userAccountsData.size() - 1 : testCaseData.userAccountsData.size();
+        try {
+            final AsyncResult<List<IAccount>> result = future.get();
+
+            if (result.getSuccess()) {
+                assertEquals("number of accounts", 1, result.getResult().size());
+                MultiTenantAccount account = (MultiTenantAccount) result.getResult().get(0);
+                assertEquals("number of tenant profiles",
+                        expectedTenantProfilesCount , account.getTenantProfiles().size());
+                assertTrue(testCaseData.homeAccountId.contains(account.getId()));
+                if(homeAccountSignedIn) {
+                    assertNotNull("account.getClaims()", account.getClaims());
+                    assertNotEquals("claims count",0, account.getClaims().size());
+                    verifyAccountDetails(testCaseData.userAccountsData.get(0), account);
+                }
+
+                int tenantProfileIndexStart = homeAccountSignedIn? 1: 0;
+                int tenantProfileIndexEnd = homeAccountSignedIn?
+                        expectedTenantProfilesCount: expectedTenantProfilesCount -1;
+                for(int i = tenantProfileIndexStart; i <= tenantProfileIndexEnd; i++) {
+                    verifyAccountDetails(
+                            testCaseData.userAccountsData.get(i),
+                            account.getTenantProfiles().get(testCaseData.userAccountsData.get(i).tenantId));
+                }
+            } else {
+                fail(result.getException().getMessage());
+            }
+        } catch (final Exception e) {
+            fail(e.getMessage());
+        }
+    }
+
+    private void verifyAccountDetails(UserAccountData expectedUserAccountData, IAccount account) {
+        assertEquals("account.getTenantId()", expectedUserAccountData.tenantId, account.getTenantId());
+        assertEquals("account.getId()", expectedUserAccountData.localAccountId, account.getId());
+        assertEquals("account.getAuthority()", expectedUserAccountData.cloud, account.getAuthority());
+        assertEquals("iss claim", expectedUserAccountData.cloud, account.getClaims().get("iss"));
+        assertEquals("tid claim", expectedUserAccountData.tenantId, account.getClaims().get("tid"));
+        assertEquals("oid claim", expectedUserAccountData.localAccountId, account.getClaims().get("oid"));
+    }
+
+    @Test
+    public void testRemoveAccountRemovesAllCredentialsFromMultipleCloudsWithSameHomeAccountId() {
+        // arrange
+        // add mock record in cache which has different home account id than the account to be removed
```

Comment:
How do we ensure this statement is always true?

Reply:
I replaced it to use a random UUID for the 2nd account. So unless we have a collision from UUID.randomUUID() (which is not likely) it should be ensured